### DISCUSSION TITLE
automatic https, am i doing it wrong?

### DISCUSSION THREAD

**Discussion by dotargz** [pocketbase/pocketbase#316] [Q&A] [answered]
I have a subdomain pointing to my pocketdb instance's IP (proxied through Cloudflare, although I also tried it without the proxy afterward and got a similar result)

This is what happens when I try to do HTTPS:
```
root@pocketdb-temp:~# ./pocketbase serve --https api.kynosocial.onespark.dev:80
> Server started at: https://api.kynosocial.onespark.dev:80
  - REST API: https://api.kynosocial.onespark.dev:80/api/
  - Admin UI: https://api.kynosocial.onespark.dev:80/_/
2022/08/15 04:17:17 http: TLS handshake error from 73.xxx.xxx.31:55007: acme/autocert: unable to satisfy "https://acme-v02.api.letsencrypt.org/acme/authz-v3/142197389387" for domain "api.kynosocial.onespark.dev": no viable challenge type found
2022/08/15 04:17:37 http: TLS handshake error from 73.xxx.xxx.31:55018: acme/autocert: missing certificate
2022/08/15 04:17:38 http: TLS handshake error from 73.xxx.xxx.31:55020: acme/autocert: missing certificate
2022/08/15 04:17:39 http: TLS handshake error from 73.xxx.xxx.31:55021: acme/autocert: missing certificate
2022/08/15 04:17:39 http: TLS handshake error from 73.xxx.xxx.31:55022: acme/autocert: missing certificate
2022/08/15 04:17:41 http: TLS handshake error from 73.xxx.xxx.31:55024: acme/autocert: missing certificate
2022/08/15 04:17:41 http: TLS handshake error from 73.xxx.xxx.31:55025: acme/autocert: missing certificate
```
(i x-ed out my IP address)

Am I doing it wrong? (also sorry for posting 2 discussions in the last 24h)

**Comment by ganigeorgiev** [comment, marked as answer]
It depends how you access your server, but the https port usually should be **443** (or 8443):
```sh
./pocketbase serve --http="api.kynosocial.onespark.dev:80" --https="api.kynosocial.onespark.dev:443"
```
_Both `--http` and `--https` are used in order to auto redirect from http to the https version of your app._

Furthermore, Let's encrypt recommends to have both 80 and 443 ports open and serving (mostly for domain validations and better usability) - https://letsencrypt.org/docs/allow-port-80/

**Comment by kewarrie** [comment]
Hi @dotargz did you ever get this to work? I'm also using Cloudflare for my subdomain, but I've only been able to get HTTP to work by running:

```
./pocketbase serve --http="0.0.0.0:80"
```

When I try to run the "serve" command with the HTTPS flag configured similarly, it fails. I'm not sure why, so I'm just wondering if you ever got a solution to this. I'm kinda stuck and out of ideas at the moment.

**Reply by dotargz** [reply]
The solution provided worked for me, but it's possible that your issue is caused by your Cloudflare SSL setting. "Flexible" makes an HTTP request to the origin, and "Full" makes an HTTPS request, so you should have the setting on "Full" if you are proxying it through Cloudflare and you have the --https flag on. Your issue could also be caused by something completely unrelated, since you didn't say any specific error messages (but if there are no errors, it's most likely your Cloudflare SSL setting).

**Reply by kewarrie** [reply]
Thanks for the prompt response. I'll provide more details later for context. In the meantime, thanks for confirming the provided solution worked for you.

**Reply by nikhil-varma-mudunuri** [reply]
hi @kewarrie i am also facing the same issue able to access via http but not https , my domain in linode, did it work for you for https

**Reply by ganigeorgiev** [reply]
@nikhil-varma-mudunuri Note that this is a very old discussion and a lot of things has changed since then (see https://github.com/pocketbase/pocketbase/discussions/3190). As mentioned in your previous post https://github.com/pocketbase/pocketbase/discussions/4027, using just `./pocketbase serve yourdomain.com` should work out of the box without the need of any extra configuration (it will bind automatically to `0.0.0.0:80` and `0.0.0.0:443`). If it doesn't work, please try to provide more details about your setup and what commands you've executed in your existing discussion or file a new one.

**Comment by greendesertsnow** [comment]
how does this work if you're serving multiple pocketbase instances on different ports for multiple projects?

**Reply by ganigeorgiev** [reply]
If you have more than one PocketBase application running on the server, then you'll be better with a reverse proxy (nginx, apache, caddy, etc) that will bind to 80/443 and will proxy the requests to your different applications. For an example nginx config, you can check https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.
